core/src/main/java/org/apache/iceberg/deletes/BitmapPositionDeleteIndex.java

```diff
@@ -46,4 +46,13 @@ public boolean isDeleted(long position) {
   public boolean isEmpty() {
     return roaring64Bitmap.isEmpty();
   }
+
+  @Override
+  public PositionDeleteIndex or(PositionDeleteIndex deleteIndex) {
+    if (deleteIndex instanceof BitmapPositionDeleteIndex) {
+      roaring64Bitmap.or(((BitmapPositionDeleteIndex) deleteIndex).roaring64Bitmap);
+      return this;
+    }
+    throw new IllegalArgumentException();
```

Comment:
This should not throw an exception with no context. Please use `Preconditions` and produce a helpful error message.

Reply:
fixed